File: src/android/MyBackgroundModePlugin.java
package com.mybackgroundmodeplugin;

import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;
import org.json.JSONArray;
import org.json.JSONException;

public class MyBackgroundModePlugin extends CordovaPlugin {
    @Override
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
        if (action.equals("enableBackgroundMode")) {
            this.enableBackgroundMode(callbackContext);
            return true;
        } else if (action.equals("disableBackgroundMode")) {
            this.disableBackgroundMode(callbackContext);
            return true;
        }
        return false;
    }

    private void enableBackgroundMode(CallbackContext callbackContext) {
        // Implement enabling background mode here
        callbackContext.success("Background mode enabled");
    }

    private void disableBackgroundMode(CallbackContext callbackContext) {
        // Implement disabling background mode here
        callbackContext.success("Background mode disabled");
    }
}

File: src/ios/MyBackgroundModePlugin.m
#import <Cordova/CDV.h>

@interface MyBackgroundModePlugin : CDVPlugin
- (void)enableBackgroundMode:(CDVInvokedUrlCommand*)command;
- (void)disableBackgroundMode:(CDVInvokedUrlCommand*)command;
@end

@implementation MyBackgroundModePlugin

- (void)enableBackgroundMode:(CDVInvokedUrlCommand*)command {
    // Implement enabling background mode here
    CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Background mode enabled"];
    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
}

- (void)disableBackgroundMode:(CDVInvokedUrlCommand*)command {
    // Implement disabling background mode here
    CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Background mode disabled"];
    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
}

@end
